app/src/main/java/com/keylesspalace/tusky/EditProfileActivity.kt

```diff
@@ -0,0 +1,491 @@
+                                            grantResults: IntArray) {
+        when (requestCode) {
+            PERMISSIONS_REQUEST_READ_EXTERNAL_STORAGE -> {
+                if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    initiateMediaPicking()
+                } else {
+                    endMediaPicking()
+                    Snackbar.make(avatarButton, R.string.error_media_upload_permission, Snackbar.LENGTH_LONG).show()
+                }
+            }
+        }
+    }
+
+    private fun initiateMediaPicking() {
+        val intent = Intent(Intent.ACTION_GET_CONTENT)
+        intent.addCategory(Intent.CATEGORY_OPENABLE)
+        intent.type = "image/*"
+        when (currentlyPicking) {
+            EditProfileActivity.PickType.AVATAR -> {
+                startActivityForResult(intent, AVATAR_PICK_RESULT)
+            }
+            EditProfileActivity.PickType.HEADER -> {
+                startActivityForResult(intent, HEADER_PICK_RESULT)
+            }
+        }
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        menuInflater.inflate(R.menu.edit_profile_toolbar, menu)
+        return super.onCreateOptionsMenu(menu)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            android.R.id.home -> {
+                onBackPressed()
+                return true
+            }
+            R.id.action_save -> {
+                save()
+                return true
+            }
+        }
+        return super.onOptionsItemSelected(item)
+    }
+
+    private fun save() {
+        if (isSaving || currentlyPicking != PickType.NOTHING) {
+            return
+        }
+
+        isSaving = true
+        saveProgressBar.visibility = View.VISIBLE
+
+        val newDisplayName = displayNameEditText.text.toString()
+        val displayName = if (oldDisplayName == newDisplayName) {
+            null
+        } else {
+            RequestBody.create(MultipartBody.FORM, newDisplayName)
+        }
+
+        val newNote = noteEditText.text.toString()
+        val note = if (oldNote == newNote) {
+            null
+        } else {
+            RequestBody.create(MultipartBody.FORM, newNote)
+        }
+
+        val avatar = if(avatarChanged) {
+            val avatarBody = RequestBody.create(MediaType.parse("image/png"), getCacheFileForName(AVATAR_FILE_NAME))
+            MultipartBody.Part.createFormData("avatar", getFileName(), avatarBody)
+        } else {
+            null
+        }
+
+        val header = if(headerChanged) {
+            val headerBody = RequestBody.create(MediaType.parse("image/png"), getCacheFileForName(HEADER_FILE_NAME))
+            MultipartBody.Part.createFormData("header", getFileName(), headerBody)
+        } else {
+            null
+        }
+
+        if(displayName == null && note == null && avatar == null && header == null) {
```

Comment:
Not clear why this check is needed. Perhaps a comment would help.

Reply:
If everything is null, nothing has changed and there is no need to make a network request :)
Will add a comment.